src/mlpack/methods/ann/layer/isru_impl.hpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * @file isru_impl.hpp
+ * @author Prince Gupta
+ *
+ * Implementation of ISRU (Inverse Square Root Unit) activation function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_ISRU_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_ISRU_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "isru.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+// This constructor is called for ISRU activation function.
+// 'alpha' is a hyperparameter
+template<typename InputDataType, typename OutputDataType>
+ISRU<InputDataType, OutputDataType>::ISRU(const double alpha) :
+    alpha(alpha)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename OutputType>
+void ISRU<InputDataType, OutputDataType>::Forward(
+    const InputType&& input, OutputType&& output)
+{
+  output = input / (arma::sqrt(1 + alpha * arma::pow(input, 2)));
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename DataType>
+void ISRU<InputDataType, OutputDataType>::Backward(
+    const DataType&& input, DataType&& gy, DataType&& g)
+{
+  double yEdge = 1 / std::sqrt(alpha);
+  double x;
+  g.set_size(arma::size(input));
+  for (size_t i = 0; i < input.n_elem; i++)
+  {
+    if (input(i) == 0)
+    {
+      g(i) = 1;
+      continue;
+    }
```

Comment:
Any particular reason why we need to use `continue` or we could go with an if - else if pair.

Reply:
`if.. else` can work too, I thought it would look cleaner with `continue`. Also if not used then  g(i) will become `nan` due to 0/0 whenever `input(i) == 0`.

Reply:
`if.. else` can work too, I thought it would look cleaner with `continue`. Also if not used then  g(i) will become `nan` due to 0/0 whenever `input(i) == 0`.

Reply:
Agreed that the condition is necessary, I think without `continue` it would also be a single line condition like the rest of them.